chain/x/zoracle/exec_env.go

```diff
@@ -63,20 +114,18 @@ func (env *ExecutionEnvironment) GetValidatorAddress(validatorIndex int64) ([]by
 	return env.request.RequestedValidators[validatorIndex], nil
 }
 
-func (env *ExecutionEnvironment) GetMaximumResultSize() int64 {
-	return int64(env.keeper.GetParam(env.ctx, KeyMaxResultSize))
-}
-
-func (env *ExecutionEnvironment) GetMaximumCalldataOfDataSourceSize() int64 {
-	return int64(env.keeper.GetParam(env.ctx, KeyMaxCalldataSize))
-}
-
 func (env *ExecutionEnvironment) RequestExternalData(
 	dataSourceID int64,
 	externalDataID int64,
 	calldata []byte,
-) error {
-	return env.keeper.AddNewRawDataRequest(env.ctx, env.requestID, types.ExternalID(externalDataID), types.DataSourceID(dataSourceID), calldata)
+) {
+	if int64(len(calldata)) <= env.maxCalldataSize {
```

Comment:
You should ignore only when  `int64(len(calldata)) > env.maxCalldataSize + 1`. That way the function is guarantee to fail down the road.

Reply:
Talk to me if you don't understand what's going on.